istioctl/cmd/istioctl/proxyconfig.go

```diff
@@ -173,13 +174,57 @@ var (
 		RunE: func(c *cobra.Command, args []string) error {
 			podName := args[0]
 			ns := handleNamespace()
-			configWriter, err := setupEnvoyConfigWriter(podName, ns, c.OutOrStdout())
+			configWriter, err := setupConfigdumpEnvoyConfigWriter(podName, ns, c.OutOrStdout())
 			if err != nil {
 				return err
 			}
 			return configWriter.PrintBootstrapDump()
 		},
 	}
+
+	clusterName       string
+	endpointConfigCmd = &cobra.Command{
+		Use:   "endpoint <pod-name>",
+		Short: "Retrieves endpoint configuration for the Envoy in the specified pod",
+		Long:  `Retrieve information about endpoint configuration for the Envoy instance in the specified pod.`,
+		Example: `  # Retrieve full endpoint configuration for a given pod from Envoy.
+  istioctl proxy-config endpoint <pod-name>
+
+  # Retrieve endpoint summary for endpoint with port 9080.
+  istioctl proxy-config endpoint <pod-name> --port 9080
+
+  # Retrieve full endpoint with a address (172.17.0.2).
+  istioctl proxy-config endpoint <pod-name> --address 172.17.0.2 -o json
+
+  # Retrieve full endpoint with a cluster name (outbound|9411||zipkin.istio-system.svc.cluster.local).
+  istioctl proxy-config endpoint <pod-name> --cluster "outbound|9411||zipkin.istio-system.svc.cluster.local" -o json
+`,
+		Aliases: []string{"ep"},
```

Comment:
Could you add endpoints as an alias as well?

Reply:
Sure, added.